metals/src/main/scala/scala/meta/internal/builds/ShellRunner.scala

```diff
@@ -87,15 +92,22 @@ class ShellRunner(time: Time, workDoneProvider: WorkDoneProgress)(implicit
   ): Future[Int] = {
     val elapsed = new Timer(time)
     val env = additionalEnv ++ JdkSources.envVariables(javaHome)
+
+    val shellArguments = userConfiguration().defaultShell match {
+      case Some(shell) => List(shell, "-i", "-l", "-c", args.mkString(" "))
```

Comment:
would all shells have the same options though?

Reply:
I really do think so. I've tried bash, zsh and fish so far, and they all seem to work with the above flags. If some obscure shell doesn't work with some of the flags, we should be able to adjust them like this.

```scala
val shellArguments = userConfiguration().defaultShell match {
  case Some(shell) =>
    shell match {
      case "/usr/bin/fish" => List(shell, "-c", args.mkString(" "))
      case _               => List(shell, "-i", "-l", "-c", args.mkString(" "))
    }
  case None => args
}
```

what do you think?

Reply:
People might have the fish shell located somewhere else, so this wouldn't work. Do we need the additional options though?

```suggestion
      case Some(shell) => shell :: args
```
should work, no?

Reply:
OR you could do `Some(shell) if shell.contains("fish")` to be on the safe side.

Reply:
We should also pack it with Try() and return an empty map if there is an exception